### ISSUE TITLE
Integration with Redux

### ISSUE THREAD

**Issue by ghengeveld** [ghengeveld/react-async#45]
Given that many React applications use Redux, integration with Redux would be beneficial for a lot of people and lower the barrier to entry.

The goal is to keep the React Async state in the global Redux store so it can be integrated with the rest of the application state.

React Async already uses a reducer internally, so integration with Redux should be fairly straightforward.

**Comment by Avi98** [comment]
hi, @ghengeveld if no one is working I can start working on it.

**Comment by ghengeveld** [comment]
Yes please, if you will. I'm currently focused on getting the DevTools released, so don't really have time to work on this.

**Comment by Avi98** [comment]
Hi @ghengeveld, Sorry I've been busy lately. I need your help on this one. I got the chance to look into redux-form's implementation. A HOC is used to connect to the passed component to redux store and then that store is being passed to the wrapped component. 

Please guide how to start with this issue

**Comment by CMCDragonkai** [comment]
I was interested in this, right now I model async state in Redux using a custom type like:

```
type Async<D, I = any, E = Error> =
  | AsyncInitial
  | AsyncProgress<I>
  | AsyncSuccess<D, I>
  | AsyncFail<I, E>;

type AsyncInitial = { type: 'Initial' };
type AsyncProgress<I = any> = { type: 'Progress'; id?: I };
type AsyncSuccess<D = any, I = any> = { type: 'Success'; id?: I; data: D };
type AsyncFail<I = any, E = Error> = { type: 'Fail'; id?: I; error: E };
```

And this is reified into redux state in order to allow components to render conditionally on async state machine.

But I wanted more control over the async type, and was hoping for a richer type that could be cancelled... etc.

**Comment by ghengeveld** [comment]
Hi @CMCDragonkai, that's one of the goals for Async Library. We want to define a rich data model for asynchronous UI state, with plenty of flexibility for enhancement, and a lot of generally useful features out of the box. That includes things like cancellation.

At the moment you can take a look / get involved in [the work that's already underway](https://github.com/async-library/react-async/pull/105) to implement React Async in TypeScript. The types you're looking for are [here](https://github.com/async-library/react-async/pull/105/files#diff-191690b17c3d84ac54dfbb570442efc7).

If you don't want to wait for the TypeScript migration to be done, you can start using the [existing types](https://github.com/async-library/react-async/blob/v8.0.0/packages/react-async/src/index.d.ts) in the library, these should be importable from `"react-async/dist-types/index.d.ts"`.

It would be very welcome if you could take a stab at integrating this with Redux.
